lib/dams/dams_object.rb

```diff
@@ -391,6 +382,23 @@ def insertFileFields( solr_doc, cid, files )
           end
 	    }
 	  end
+      def fulltext( object, file, cid )
+        mime_type = file.mimeType.first.to_s.gsub(/;.*/,"")
+        if ["application/pdf", "text/html", "text/plain"].include?(mime_type)
+          fid = (cid != nil) ? "_#{cid}_#{file.id}" : "_" + file.id
+          fid = "fulltext#{fid}" if mime_type == "application/pdf"
```

Comment:
Thanks, makes perfect sense. Do you think it would be easier to come back and refactor/migrate this in the future if the fulltext extraction conditional assignment was a method that described the requirement? Something like:
```
add_fulltext_service_trigger(fid) if mime_type == "application/pdf"
.....
# Add a prefix to PDF file to trigger repository text extraction service
def add_fulltext_service_trigger(fid)
  "fulltext#{fid}"
end
```

Or, do you think we're moving far enough away from this via Sufia/F4 that it's not worth the hassle and refactoring/migrating won't be an issue?